### PR TITLE
 Inelastic Settings option to turn off Load Workspace History 

### LINKED ISSUE #37691: Inelastic Settings option to turn off Load Workspace History
**Description**
Loading the workspace history for a nexus file is often the slowest part of loading a file. It might be worth having an option in the Inelastic Settings Dialog that can be used to turn off Loading workspace history. This option can then be used for all DataSelectors and AddWorkspaceDialogs in the Inelastic/Indirect interfaces.

[LoadNexusProcessed ](https://docs.mantidproject.org/nightly/algorithms/LoadNexusProcessed-v2.html) has a "LoadHistory" option.

Default might be to have this "LoadHistory" as true. We can then let users know that the option is available and could lead to faster loading speeds if they do not care about loading workspace history.

**Acceptance Criteria**
- An option to turn off/on Loading workspace history is available in the Inelastic settings dialog
- This option is used by all the Data Selectors/AddWorkspaceDialogs in the Inelastic/Indirect interfaces
- Make sure to update Settings documentation

### PR DESCRIPTION
**Pull request by adriazalvarez**
### Description of work
This PR adds an option to the Indirect Settings dialog to load the history of a workspace, for details see #37691 . This will allow for faster workflows as sometimes loading the history can slog the loading algorithm #36769. 
This setting is now available in most indirect/inelastic interfaces that preload data using 'LoadNexusProcessed' algorithm to the ADS (as the ones having data selectors or workspace dialogs for selecting input data). It won't be available for loaders that load raw files (like in diffraction or data reduction interfaces). 

Fixes #37691 . 

### To test:
As there are many affected interfaces, the instructions are slightly tedious, I'm sorry for that. 

- First, make sure to delete the `[Indirect Settings]` and any setting in that group from the `mantidworkbench.ini` file that stores the settings in your machine. (You can find 'mantidworkbench.ini' either in `%APPDATA/../mantidproject` on Windows or in `.config/mantidproject` on MacOS  and Linux.) . Then save the file. 
- Open Mantid Workbench, and then any Indirect/Inelastic interface. Once on the interface, click on the cog icon for the settings, which should be located at the bottom left corner of the interface. The settings dialog will open. 
- On the settings dialog, the options: `Restrict allowed input files by name` and `Load Workspace History` should be ticked by default. Close the dialog. 

Now, for every Inelastic Interface, do the following for each tab (except  on `Calculate Paalman Pings` which is deprecated): 

- Open the settings dialog and tick on the `Load Workspace History` setting. 
- Load sample data for the respective interface (either with a dialog, or from the data selector)
- Once the sample data is loaded onto the ADS, right-click on the workspace and select `Show History`. You should see the history log for that workspace. 
- Clear the ADS. 
- Open the Settings dialog again and untick the `Load Workspace History` setting.
- Load the same sample data as before. 
- Once the sample data is loaded onto the ADS, right-click on the workspace and select `Show History`. You should see only 1 entry on the history log (the load algorithm). 
- Move onto the next interface. 

For the Indirect Interfaces, it is only necessary to do the loading history check with the Calibration Data Selector of the `ISIS Energy Transfer` and `ISIS Diagnostics` tabs (the one allowing to load calibration files) and with the Transmission tab. 

Finally, build the docs and check that the `Interfaces/Indirect Settings` section contains and entry for the load history settings. Check that is well formatted and understandable. 

*A release note has been added and the documentation entry for settings dialog updated*

---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.

### CODE REVIEW

**Review by robertapplin — CHANGES_REQUESTED** [review]
This is a really useful new feature, thanks for the implementation! Just spotted a few potential typos in the code

**Comment by robertapplin** [comment]
This is working really well - loading on the QENS Fitting interface in particular is much faster when not loading history. I have tested on all Indirect/Inelastic interfaces and it is working as expected. Just one place where it may not be working is on the Indirect Simulations interface->MolDyn tab for the Resolution input. Do you see this too?

**Comment by adriazalvarez** [comment]
You are right, the simulations interface was missing. Thanks for catching that up. I have added it now.

File: qt/scientific_interfaces/Indirect/Simulation/SimulationTab.h [hunk 1]
```diff
@@ -39,9 +39,11 @@ class MANTIDQT_INDIRECT_DLL SimulationTab : public InelasticTab {
   void setOutputPlotOptionsWorkspaces(std::vector<std::string> const &outputWorkspaces);
   void clearOutputPlotOptionsWorkspaces();
 
+  void enableLoadHistoryProperty(bool doLoadHistory);
   virtual void loadSettings(const QSettings &settings) = 0;
 
 private:
+  virtual void setLoadHistory(bool doLoadHistory){UNUSED_ARG(doLoadHistory)};
```
Comment (robertapplin) [inline qt/scientific_interfaces/Indirect/Simulation/SimulationTab.h:46]:
```suggestion
  virtual void setLoadHistory(bool doLoadHistory) { UNUSED_ARG(doLoadHistory); }
```

**Review by robertapplin — APPROVED** [review]
This is a really nice addition to the Indirect/Inelastic settings. If users are finding it slow to load data into one of our interfaces, they now have the option to untick the "Load Workspace History" option. In some cases (such as on QENS Fitting), the time it takes to load certain datasets with this option is much reduced!

**Comment by rbauststfc** [comment]
Just a quick question before merging this in, I didn't notice many changes/additions to tests in this PR. Are we happy that it's not possible/useful to add any more test coverage, for example where we have added new methods to a presenter (I know we can't do it for additions to the views)?

**Comment by adriazalvarez** [comment]
Thanks for your suggestions. 
I don't think I can add much more test cases on the settings presenter regarding this setting.
The implementation of the settings is done on QObjects on interfaces which are not tested at this moment. 
If you agree, I think we can move forward with the merge.

**Pull request merged**